test/viper/ok/reverse.vpr.ok

```diff
@@ -116,12 +116,18 @@ method copy_xarray($Self: Ref)
       var i: Int
       length := $size(($Self).xarray);
       inhale $array_acc(t, $int, write);
-      inhale ($size(t) == 5);
-      ($loc(t, 0)).$int := 0;
-      ($loc(t, 1)).$int := 0;
-      ($loc(t, 2)).$int := 0;
-      ($loc(t, 3)).$int := 0;
-      ($loc(t, 4)).$int := 0;
+      inhale ($size(t) == length);
+      { var $i: Int
+         $i := 0;
+         while (($i < length))
+            invariant ($i >= 0)
+            invariant $array_acc(t, $int, write)
+            { 
+              assert ($i < length);
+              ($loc(t, $i)).$int := 0;
+              $i := ($i + 1); 
+            }; 
+       };
```

Comment:
```suggestion
      inhale ($size(t) == length);
      inhale forall i : Int :: 0 <= i < $size(t) ==> $loc(t, i).$int == 0
```
So it will be equivalent to call Viper method
```
method Array_init<T>(length: Int, val: T) returns ($res : Array)
  requires length >= 0
  ensures $array_acc($res, type_field(T), write)
  ensures forall i : Int :: 0 <= i < $size(t) ==> $loc(t, i).$int == val
```
I would just add it to prelude, but it's parameterized by `T` so we should generate it. I think in a future it could be even generated from Motoko by special annotation that drop body and then monomorphisation do its job.

Now probably I would just remove `array_init_const`

Reply:
I decided to do it with a macro

```
define $array_init(a, t, x) forall i : Int :: 0 <= i && i < $size(a) ==> $loc(a, i).t == x
```

Reply:
I think it's okay for now.